tessera-jaxrs/sync-jaxrs/src/main/java/com/quorum/tessera/p2p/RestP2pClient.java

```diff
@@ -0,0 +1,62 @@
+package com.quorum.tessera.p2p;
+
+import com.quorum.tessera.partyinfo.P2pClient;
+import com.quorum.tessera.partyinfo.ResendRequest;
+import java.util.Objects;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+public class RestP2pClient implements P2pClient {
+
+    private final Client client;
+
+    public RestP2pClient(Client client) {
+        this.client = Objects.requireNonNull(client);
+    }
+
+    @Override
+    public byte[] push(String targetUrl, byte[] data) {
+
+        final Response response =
+                client.target(targetUrl)
+                        .path("/push")
+                        .request()
+                        .post(Entity.entity(data, MediaType.APPLICATION_OCTET_STREAM_TYPE));
+
+        if (Response.Status.OK.getStatusCode() != response.getStatus()
```

Comment:
Is it worth allowing any type of successful request? If we aren't limiting to exactly one status code, we could allow the entire family with `Response.Status.Family.SUCCESSFUL.equals(response.getStatusInfo().getFamily());`